testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -446,6 +446,162 @@ public void testTrinoReadsSparkPartitionedTable(StorageFormat storageFormat, int
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST, ICEBERG_JDBC}, dataProvider = "storageFormats")
+    public void testSparkReadsTrinoNestedPartitionedTable(StorageFormat storageFormat)
+    {
+        String baseTableName = toLowerCase("test_spark_reads_trino_nested_partitioned_table_" + storageFormat + randomNameSuffix());
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery(format(
+                "CREATE TABLE %s (_string VARCHAR, _varbinary VARBINARY, _bigint BIGINT, _struct ROW(_field INT, _another_field VARCHAR))" +
+                        " WITH (partitioning = ARRAY['\"_struct._field\"'], format = '%s')",
+                trinoTableName,
+                storageFormat));
+        onTrino().executeQuery(format(
+                "INSERT INTO %s VALUES" +
+                        " ('update', X'0ff102f0feff', 1001, ROW(1, 'x'))," +
```

Comment:
What is the relevance of the `varbinary` field in this test? I'm particularly interested why `varbinary` and not something easier to digest like `varchar` or `int`